src/app/startgame/page.tsx

```diff
@@ -0,0 +1,175 @@
+        break;
+      }
+
+      if (matches === 3) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+function hasTied(slots: number[]) {
+  return slots.length === 9 && !hasPlayerWon(slots);
+}
+
+function MarkIcon({ mark }: { mark: EMark }) {
+  if (mark === EMark.CROSS) {
+    return <Image src={cross} alt="cross" width="70" height="70" />;
+  }
+  return <Image src={nought} alt="nought" width="70" height="70" />;
+}
+
+export default function StartGame() {
+  const { players, setPlayers } = useContext(PlayersContext);
+  const { gameState, setGameState } = useContext(GameStateContext);
+  const { gameScore, setGameScore } = useContext(GameScoreContext);
+  const [currentPlayer, setCurrentPlayer] = useState<"p1" | "p2">(
+    players.p1.mark === EMark.CROSS ? "p1" : "p2"
+  );
+  const modalOptions = useGetModalOptions(gameState, setCurrentPlayer);
+  const cellDesign = Array.from({ length: 9 }, (_, index) => index + 1);
+
+  function handleOnClick(slot: number) {
+    const copyPlayers = { ...players };
+
+    copyPlayers[currentPlayer].slots.push(slot);
+    if (hasPlayerWon(players[currentPlayer].slots)) {
+      setGameState(
+        currentPlayer === "p1"
+          ? GameState.FIRST_PLAYER_WIN
+          : GameState.SECOND_PLAYER_WIN
+      );
+
+      if (currentPlayer === "p1") {
+        setGameScore({
+          ...gameScore,
+          p1: gameScore.p1 + 1,
+        });
+      }
+      if (currentPlayer === "p2") {
+        setGameScore({
+          ...gameScore,
+          p2: gameScore.p2 + 1,
+        });
+      }
+    }
+    if (hasTied([...players.p1.slots, ...players.p2.slots])) {
+      setGameState(GameState.TIED);
+      setGameScore({
+        ...gameScore,
+        ties: gameScore.ties + 1,
+      });
+    }
+    setPlayers(copyPlayers);
+    setCurrentPlayer(currentPlayer === "p1" ? "p2" : "p1");
+  }
+
+  return (
+    <div className="flex flex-col justify-center gap-10 mx-auto w-1/2 md:w-1/4 h-screen">
+      <Modal
+        gameResult={modalOptions?.gameResult || ""}
+        firstBtnLabel={modalOptions?.firstBtnLabel || ""}
+        secondBtnLabel={modalOptions?.secondBtnLabel || ""}
+        showModal={modalOptions !== null}
+        navigate={modalOptions?.navigate || ""}
+        onFirstBtnClick={modalOptions?.onFirstBtnClick || (() => {})}
+        onSecondBtnClick={modalOptions?.onSecondBtnClick || (() => {})}
+      />
+      <div className="flex justify-between items-end">
+        <Image src={logo} alt="logo" className="h-10 w-20" />
+        <TurnLogo currentPlayerMark={players[currentPlayer].mark} />
+        <RestartButton
+          onRestart={() => {
+            setGameState(GameState.RESTART);
+            setPlayers({
+              p1: { mark: EMark.CROSS, slots: [] },
+              p2: { mark: EMark.NOUGHT, slots: [] },
+            });
+          }}
+        />
+      </div>
+      <div className="grid grid-cols-3 gap-4 justify-between">
+        {cellDesign.map((_, index) => {
```

Comment:
Let's move this into its own component. So the levels of abstraction is the same here.

Like this component is a higher overview, smart component, no too many details

Reply:
gotta need help here! I tried to do it but the <MarkIcon /> component wasn't working properly